### ISSUE TITLE
bug(linter): false positive in `consistent-function-scoping` in `AssignmentExpression`s

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5159]
This snippet should not violate `unicorn/consistent-function-scoping`, but it currently does:

```js
module.exports = function foo() {}
```

**Comment by camc314** [comment]
> @camc314 you working on this?

not currently. feel free to do it

**Comment by Arian94** [comment]
@camc314 alright. tnx.
